app/src/main/kotlin/co/touchlab/droidconandroid/EventDetailAdapter.kt

```diff
@@ -69,39 +59,38 @@ class EventDetailAdapter(val context: Context, val frag:EventDetailFragment, val
 
     //=================== Adapter Overrides ===================
     override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): RecyclerView.ViewHolder? {
-        var holder: RecyclerView.ViewHolder? = null
         when (viewType) {
             TYPE_HEADER -> {
                 val view = LayoutInflater.from(context).inflate(R.layout.item_event_header, parent, false)
-                holder = HeaderVH(view)
+                return HeaderVH(view)
```

Comment:
OPTIONAL: Could put a single return in front of the "when" and it will return the last line from the case it picks. I could see that being slightly less readable though, so take it or leave it